### ISSUE TITLE
VT api, scrolling expected to be part of animation?

### ISSUE THREAD

**Issue by illispi** [withastro/astro#9569]
### Astro Info

```block
Astro                    v4.0.5
Node                     v20.8.1
System                   Linux (x64)
Package Manager          pnpm
Output                   static
Adapter                  none
Integrations             @astrojs/mdx
                         @astrojs/solid-js
                         @astrojs/tailwind
```

### If this issue only occurs in one browser, which browser is a problem?

Chrome

### Describe the Bug

Is view transition api expected to add "scrolling" into part of animation on page transition.

https://delvis.org/blog , click on first blog, scroll down a bit and hit back. If you slow down animation in dev tools you can see that it has added scrolling to be part of the animation. Here is the code I used, but it happens on all presets too, unless left empty.

```
  const anim: TransitionAnimationPair = {
    old: {
      name: "slideOutFade",
      duration: "0.3s",
      easing: "ease-in",
      fillMode: "both",
    },
    new: {
      name: "slideInFade",
      delay: "0.3s",
      duration: "0.3s",
      easing: "ease-out",
      fillMode: "both"
    },
  };
  
  export const slideFade = {
    forwards: anim,
    backwards: anim,
  };

@keyframes slideOutFade {
  from {
    transform: translateX(0%);
    opacity: 100%;
  }

  to {
    transform: translateX(-100px);
    opacity: 0%;
  }
}
@keyframes slideInFade {
  from {
    transform: translateX(100px);
    opacity: 0%;
  }

  to {
    transform: translateX(0%);
    opacity: 100%;
  }
}
```

### What's the expected result?

In firefox without VT api, it correctly doesn't scroll to top when hitting back on scrolled down blog post.

### Link to Minimal Reproducible Example

https://github.com/illispi/homeSite

### Participation

- [x] I am willing to submit a pull request for this issue.

**Comment by martrapp** [comment]
The scrolling is most likely the effect of the group/morph animation in Chrome. It identifies elements with the same transition names on the old and the new page. In your case, the root element and the `<main>` element, which Astro has assigned a synthetic view transition name of `astro-2g2b4d5l-1`. You can set the names explicitly with `transition:name=...`. The built-in animation treats old/main and new/main as rectangles. It gradually adjusts the size of the old/rectangle to that of the new one and simultaneously moves it to the coordinates of the new main. While this is happening, the image of the old/main element is faded out and the image of the new/main element is faded in at the same time. Since your main element covers most of the page, it looks like the page is scrolling. But you can see that the scroll bar does not move. 


Besides the in and out animations that are controlled with `::view-transition-new` and `::view-transition-old` and for which you showed the animation definitions above, you also have two of those morph/group animations on your page:
![image](https://github.com/withastro/astro/assets/94928215/a9552bf8-dde9-4f19-a230-46e8a46ba9f1)
Both are browser generated.

They are automatically generated by Chrome. If you want to get rid of them, try this
```
<style is:global>
	::view-transition-group(*) {
		animation: none;
	}
</style>
```

**Comment by martrapp** [comment]
@illispi Saw your thumbs up emoji. Did it work for you? Do you want to close this issue or is there anything else to do?

**Comment by illispi** [comment]
Sorry, I tried your code and more but that simply jumps to top of the page and then animates. Instead of animating to side and fading from the place it's currently.

**Comment by martrapp** [comment]
no problem. I'm assume you want to achieve a special behavior with view transitions that? Maybe it would be better to open a #support thread at discord?

**Comment by illispi** [comment]
I did open on discord support, but at least I understand the problem better. It does work like how I want on firefox, but I guess it's not a priority since custom implementation by astro is just until other browsers catch up with vt api. But I think you can close this since it doesn't appear to be a bug, just the way vt api works right now. Maybe it will improve once vt api gets native mpa support :)

**Comment by martrapp** [comment]
Emulation on Firefox is not identical to what the native implementation does. The main differences are:
- The morph animations are not supported. 
- The old page isn't "screenshoted", you still see the "live view"
- While Chrome can use that screenshot to run old and new animation in parallel after the DOM was changes "behind" the scenes", Firefox emulation does:  old animation, DOM swap, new animation
So Firefox simulation runs twice as long

If we can look on the Firefox version (discussion on discord) we might be able to create what you want on Chrome, too. :)
